feat(bitbucketdatacenter): support for pull request file changes

henkka opened the pull request:
Closes #1138.

This pull request brings the missing "Get pull request file changes" functionality for the Bitbucket Datacenter forge.

The `replace` directive from the `go.mod` shall be removed once https://github.com/neticdk/go-bitbucket/pull/17 is merged and new version released in the `neticdk/go-bitbucket` library, but I figured I might open this pull request already for review.

6543 commented:
looks good :) - I'll try to wait for upstream sdk to get it merged first

through i did run the CI so we  can see if it would pass already :rocket:

henkka commented:
> looks good :) - Let us try to wait for upstream sdk to get it merged first
> 
> through i did run the CI so we can see if it would pass already 🚀

@6543 in collaboration with @langecode we got the changes released in the upstream sdk, so I removed the `replace` directive and updated the sdk version in https://github.com/woodpecker-ci/woodpecker/pull/5205/commits/90257ad475c30da19eb50ec5d52f804a7278dbc5

anbraten reviewed (approved):
lgtm thanks for the work

6543 commented:
thx :)
